> This would need more work as the NE10 library is very outdated and cannot be compiled with current ct-ng toolchain.
> 
> I also suspect the low performance of libopus is actually due to compiler/env bugs in libopus, as the current artifacts generated is still with slow performance.

https://github.com/xiph/opus/blob/7db26934e4156597cb0586bb4d2e44dccdde1a59/src/opus_decoder.c#L37
```c
#if defined(__GNUC__) && (__GNUC__ >= 2) && !defined(__OPTIMIZE__) && !defined(OPUS_WILL_BE_SLOW)
# pragma message "You appear to be compiling without optimization, if so opus will be very slow."
#endif
```

// macos/arm64
```
2024-12-20T07:37:15.7634410Z configure:
2024-12-20T07:37:15.7637110Z ------------------------------------------------------------------------
2024-12-20T07:37:15.7639140Z   opus unknown:  Automatic configuration OK.
2024-12-20T07:37:15.7639450Z 
2024-12-20T07:37:15.7639630Z     Compiler support:
2024-12-20T07:37:15.7640110Z 
2024-12-20T07:37:15.7640340Z       C99 var arrays: ................ yes
2024-12-20T07:37:15.7741050Z       C99 lrintf: .................... yes
2024-12-20T07:37:15.7841850Z       Use alloca: .................... no (using var arrays)
2024-12-20T07:37:15.7847330Z 
2024-12-20T07:37:15.7854220Z     General configuration:
2024-12-20T07:37:15.7854430Z 
2024-12-20T07:37:15.7854660Z       Floating point support: ........ yes
2024-12-20T07:37:15.7854990Z       Fast float approximations: ..... yes
2024-12-20T07:37:15.7855330Z       Fixed point debugging: ......... no
2024-12-20T07:37:15.7855790Z       Inline Assembly Optimizations: . No inline ASM for your platform, please send patches
2024-12-20T07:37:15.7856240Z       External Assembly Optimizations: 
2024-12-20T07:37:15.7856650Z       Intrinsics Optimizations: ...... ARM (NEON) (NE10) (NEON Aarch64) (DOTPROD) (DOTPROD Aarch64)
2024-12-20T07:37:15.7857030Z       Run-time CPU detection: ........ no
2024-12-20T07:37:15.7857370Z       Custom modes: .................. no
2024-12-20T07:37:15.7857750Z       Assertion checking: ............ no
2024-12-20T07:37:15.7858290Z       Hardening: ..................... yes
2024-12-20T07:37:15.7858610Z       Fuzzing: ....................... no
2024-12-20T07:37:15.7858830Z       Check ASM: ..................... no
2024-12-20T07:37:15.7859470Z 
2024-12-20T07:37:15.7859820Z       API documentation: ............. yes
2024-12-20T07:37:15.7860070Z       Extra programs: ................ no
2024-12-20T07:37:15.7860350Z ------------------------------------------------------------------------

...

2024-12-20T07:37:20.9538410Z src/opus_decoder.c:37:10: warning: You appear to be compiling without optimization, if so opus will be very slow. [-W#pragma-messages]
2024-12-20T07:37:20.9558960Z # pragma message "You appear to be compiling without optimization, if so opus will be very slow."
```


// linux/arm64
```
2024-12-20T06:03:00.7064137Z #78 31.51 configure:
2024-12-20T06:03:00.7064615Z #78 31.51 ------------------------------------------------------------------------
2024-12-20T06:03:00.7065210Z #78 31.52   opus unknown:  Automatic configuration OK.
2024-12-20T06:03:00.7065648Z #78 31.52 
2024-12-20T06:03:00.7066173Z #78 31.52     Compiler support:
2024-12-20T06:03:00.7066506Z #78 31.52 
2024-12-20T06:03:00.7066803Z #78 31.52       C99 var arrays: ................ yes
2024-12-20T06:03:00.7067265Z #78 31.52       C99 lrintf: .................... yes
2024-12-20T06:03:00.7067932Z #78 31.52       Use alloca: .................... no (using var arrays)
2024-12-20T06:03:00.7068413Z #78 31.52 
2024-12-20T06:03:00.7068695Z #78 31.52     General configuration:
2024-12-20T06:03:00.7069041Z #78 31.52 
2024-12-20T06:03:00.7069355Z #78 31.52       Floating point support: ........ yes
2024-12-20T06:03:00.7069830Z #78 31.52       Fast float approximations: ..... yes
2024-12-20T06:03:00.7070309Z #78 31.52       Fixed point debugging: ......... no
2024-12-20T06:03:00.7071059Z #78 31.52       Inline Assembly Optimizations: . No inline ASM for your platform, please send patches
2024-12-20T06:03:00.7071774Z #78 31.52       External Assembly Optimizations: 
2024-12-20T06:03:00.7072428Z #78 31.52       Intrinsics Optimizations: ...... ARM (NEON) (NEON Aarch64) (DOTPROD)
2024-12-20T06:03:00.7073160Z #78 31.52       Run-time CPU detection: ........ ARM (DOTPROD Intrinsics)
2024-12-20T06:03:00.7073727Z #78 31.52       Custom modes: .................. no
2024-12-20T06:03:00.7074190Z #78 31.52       Assertion checking: ............ no
2024-12-20T06:03:00.7074652Z #78 31.52       Hardening: ..................... yes
2024-12-20T06:03:00.7075098Z #78 31.52       Fuzzing: ....................... no
2024-12-20T06:03:00.7075521Z #78 31.52       Check ASM: ..................... no
2024-12-20T06:03:00.7075917Z #78 31.52 
2024-12-20T06:03:00.7076231Z #78 31.52       API documentation: ............. yes
2024-12-20T06:03:00.7076676Z #78 31.52       Extra programs: ................ no
2024-12-20T06:03:00.7077357Z #78 31.52 ------------------------------------------------------------------------
```

// linux/amd64
```
2024-12-20T06:09:43.2431650Z #73 32.88 configure:
2024-12-20T06:09:43.2432054Z #73 32.88 ------------------------------------------------------------------------
2024-12-20T06:09:43.2432609Z #73 32.88   opus unknown:  Automatic configuration OK.
2024-12-20T06:09:43.2433042Z #73 32.88 
2024-12-20T06:09:43.2433319Z #73 32.88     Compiler support:
2024-12-20T06:09:43.2433661Z #73 32.88 
2024-12-20T06:09:43.2433944Z #73 32.88       C99 var arrays: ................ yes
2024-12-20T06:09:43.2434376Z #73 32.88       C99 lrintf: .................... yes
2024-12-20T06:09:43.2434860Z #73 32.88       Use alloca: .................... no (using var arrays)
2024-12-20T06:09:43.2435324Z #73 32.88 
2024-12-20T06:09:43.2435605Z #73 32.88     General configuration:
2024-12-20T06:09:43.2435959Z #73 32.88 
2024-12-20T06:09:43.2436254Z #73 32.88       Floating point support: ........ yes
2024-12-20T06:09:43.2436727Z #73 32.88       Fast float approximations: ..... yes
2024-12-20T06:09:43.2437190Z #73 32.88       Fixed point debugging: ......... no
2024-12-20T06:09:43.2437884Z #73 32.88       Inline Assembly Optimizations: . No inline ASM for your platform, please send patches
2024-12-20T06:09:43.2438606Z #73 32.88       External Assembly Optimizations: 
2024-12-20T06:09:43.2439160Z #73 32.88       Intrinsics Optimizations: ...... x86 SSE SSE2 SSE4.1
2024-12-20T06:09:43.2439706Z #73 32.88       Run-time CPU detection: ........ x86 SSE4.1
2024-12-20T06:09:43.2440178Z #73 32.88       Custom modes: .................. no
2024-12-20T06:09:43.2440619Z #73 32.88       Assertion checking: ............ no
2024-12-20T06:09:43.2441060Z #73 32.88       Hardening: ..................... yes
2024-12-20T06:09:43.2441494Z #73 32.88       Fuzzing: ....................... no
2024-12-20T06:09:43.2441921Z #73 32.88       Check ASM: ..................... no
2024-12-20T06:09:43.2442300Z #73 32.88 
2024-12-20T06:09:43.2442609Z #73 32.88       API documentation: ............. yes
2024-12-20T06:09:43.2443056Z #73 32.88       Extra programs: ................ no
2024-12-20T06:09:43.2443561Z #73 32.89 ------------------------------------------------------------------------
```

--- a/builder/scripts.d/45-libNE10.sh
+++ b/builder/scripts.d/45-libNE10.sh
@@ -1,7 +1,7 @@
 #!/bin/bash
 
-SCRIPT_REPO="https://github.com/projectNe10/Ne10.git"
-SCRIPT_COMMIT="1f059a764d0e1bc2481c0055c0e71538470baa83"
+SCRIPT_REPO="https://github.com/gnattu/Ne10.git"
+SCRIPT_COMMIT="545f4f18014cdbf9fb5fb1a9f5d24000200dfa8b"
 
 ffbuild_enabled() {
     [[ $TARGET == win* ]] && return -1

--- a/builder/scripts.d/50-libopus.sh
+++ b/builder/scripts.d/50-libopus.sh
@@ -40,7 +40,13 @@ ffbuild_dockerbuild() {
 
     # reset CLFAGS because libopus may give up optimization if current CFLAGS contains value
     CFLAGS_BACKUP="$CFLAGS"
-    export CFLAGS="-O3" # For some reason libopus will not add optimization flag, we have to set it ourselves
+
+    # For some reason libopus will not add optimization flag, we have to set it ourselves
+    if [[ $TARGET == mac* ]]; then
+          export CFLAGS="-O3"
+      else
+          export CFLAGS="-O3 -fPIC -DPIC"
+      fi
 
     ./configure "${myconf[@]}"
     make -j$(nproc)

--- a/builder/scripts.d/50-libopus.sh
+++ b/builder/scripts.d/50-libopus.sh
@@ -38,9 +38,15 @@ ffbuild_dockerbuild() {
         )
     fi
 
+    # reset CLFAGS because libopus may give up optimization if current CFLAGS contains value
+    CFLAGS_BACKUP="$CFLAGS"
+    export CFLAGS="-O3" # For some reason libopus will not add optimization flag, we have to set it ourselves
+
     ./configure "${myconf[@]}"
     make -j$(nproc)
     make install
+
+    export CFLAGS="$CFLAGS_BACKUP"
 
     if [[ $TARGET == *arm64 ]]; then
         if [[ $TARGET == mac* ]]; then

